### PR TITLE
Issue-835: Upgrade to pac4j v1.6.0

### PR DESCRIPTION
**Pull request by leleuj**

### CODE REVIEW

File: cas-server-support-pac4j/src/main/java/org/jasig/cas/support/pac4j/web/flow/ClientAction.java [hunk 1]
```diff
@@ -189,9 +192,8 @@ protected void prepareForLoginPage(final RequestContext context) {
 
         // save parameters in web session
         final Service service = (Service) context.getFlowScope().get(SERVICE);
-        if (service != null) {
-            session.setAttribute(SERVICE, service);
-        }
+        logger.debug("save service: {}", service);
```
Comment (leleuj) [inline cas-server-support-pac4j/src/main/java/org/jasig/cas/support/pac4j/web/flow/ClientAction.java]: This is a fix for a small bug I found when testing: when you play a first authentication for a service and then a second one without a service, it was not working as the ```null``` service was not saved (and thus restored).

File: cas-server-support-pac4j/pom.xml [hunk 2]
```diff
@@ -60,10 +60,17 @@
 			<scope>test</scope>
 			<version>${pac4j.version}</version>
 		</dependency>
+
+        <dependency>
+            <groupId>org.pac4j</groupId>
+            <artifactId>pac4j-http</artifactId>
+            <scope>test</scope>
+            <version>${pac4j.version}</version>
```
Comment (mmoayyed) [inline cas-server-support-pac4j/pom.xml]: Looks like we have an indentation issue here

File: cas-server-support-pac4j/src/main/java/org/jasig/cas/support/pac4j/web/flow/ClientAction.java [hunk 3]
```diff
@@ -129,20 +129,22 @@ protected Event doExecute(final RequestContext context) throws Exception {
...
                     .findClient(clientName);
-            logger.debug("client : {}", client);
+            logger.debug("client: {}", client);
 
-            // HTTP protocol not allowed
-            if (client.getProtocol() == Protocol.HTTP) {
-                throw new TechnicalException("HTTP protocol client not supported : " + client);
+            // Allow CAS, OAuth, OpenID and SAML protocols
+            final Mechanism mechanism = client.getMechanism();
+            if (mechanism != Mechanism.CAS_PROTOCOL && mechanism != Mechanism.OAUTH_PROTOCOL
```
Comment (mmoayyed) [inline cas-server-support-pac4j/src/main/java/org/jasig/cas/support/pac4j/web/flow/ClientAction.java]: Suggestion: Create a list of all supported protocols as a const, and do `list.contains(mechanism)=false` instead.

File: cas-server-support-pac4j/src/main/java/org/jasig/cas/support/pac4j/web/flow/ClientAction.java [hunk 4]
```diff
@@ -189,9 +192,8 @@ protected void prepareForLoginPage(final RequestContext context) {
 
         // save parameters in web session
         final Service service = (Service) context.getFlowScope().get(SERVICE);
```
Comment (mmoayyed) [inline cas-server-support-pac4j/src/main/java/org/jasig/cas/support/pac4j/web/flow/ClientAction.java]: Dont we have a util function for this, in WebUtils?

File: cas-server-support-pac4j/pom.xml (hunk 2 above)
Comment (leleuj) [inline cas-server-support-pac4j/pom.xml]: Yes. Will fix.

File: cas-server-support-pac4j/src/main/java/org/jasig/cas/support/pac4j/web/flow/ClientAction.java (hunk 3 above)
Comment (leleuj) [inline cas-server-support-pac4j/src/main/java/org/jasig/cas/support/pac4j/web/flow/ClientAction.java]: Good idea. Will do.

File: cas-server-support-pac4j/src/main/java/org/jasig/cas/support/pac4j/web/flow/ClientAction.java (hunk 4 above)
Comment (leleuj) [inline cas-server-support-pac4j/src/main/java/org/jasig/cas/support/pac4j/web/flow/ClientAction.java]: Yes, we have a method for that. I will use it instead.

**Comment by leleuj** [comment]
Updated according to @mmoayyed's feedbacks.

Will merge on Monday unless someone has a new concern.

**Pull request merged**